cms/bundles/views/preview.py

```diff
@@ -66,3 +66,59 @@ def get(self, request: "HttpRequest", *args: Any, **kwargs: Any) -> TemplateResp
         )
 
         return TemplateResponse(request, page.get_template(request), context)
+
+
+class PreviewBundleReleaseCalendarView(TemplateView):
+    http_method_names: Sequence[str] = ["get"]
+
+    def get(self, request: "HttpRequest", *args: Any, **kwargs: Any) -> TemplateResponse:
+        bundle_id = kwargs["bundle_id"]
+        bundle = get_object_or_404(Bundle, id=bundle_id)
+
+        release_calendar_page = bundle.release_calendar_page
+
+        if not release_calendar_page:
+            raise Http404
+
+        log_data_entry = {
+            "type": "calendar",
+            "id": release_calendar_page.id,
+            "title": getattr(release_calendar_page, "display_title", release_calendar_page.title),
+        }
+
+        if not user_can_preview_bundle(request.user, bundle):
+            log(
+                action="bundles.preview.attempt",
+                instance=bundle,
+                data=log_data_entry,
+            )
+            raise PermissionDenied
+
+        # Make adjustments to page for preview
+        release_calendar_page.status = ReleaseStatus.PUBLISHED
+        release_calendar_page.content = cast(
+            StreamField, serialize_bundle_content_for_preview_release_calendar_page(bundle, self.request.user)
+        )
+        release_calendar_page.datasets = cast(StreamField, serialize_datasets_for_release_calendar_page(bundle))
+
+        context = release_calendar_page.get_context(request)
+
+        log(
+            action="bundles.preview",
+            instance=bundle,
+            data=log_data_entry,
+        )
+
+        if user_can_manage_bundles(self.request.user):
+            pages_in_bundle = bundle.get_bundled_pages(specific=True)
+        else:
+            pages_in_bundle = bundle.get_pages_for_previewers()
+
+        request.is_dummy = True  # type: ignore[attr-defined]
+        request.is_preview = True  # type: ignore[attr-defined]
+        request.preview_mode = "bundle-preview"  # type: ignore[attr-defined]
+
+        context["bundle_inspect_url"] = reverse("bundle:inspect", args=[bundle_id])
+        context["preview_items"] = get_preview_items_for_bundle(bundle, release_calendar_page.id, pages_in_bundle)
```

Comment:
This needs a test. tl;dr when getting the rendered preview, we have the bundle_inspect_Url and the expected `preview_items` in the response context.

Other than that LGTM